### PR TITLE
[Tables] GA review

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Review by seankane-msft — APPROVED** [review]
small comments

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 1]
```diff
@@ -484,18 +467,18 @@ class TableItem(object):
...
         :param str name: Name of the Table
         """
         self.name = name
 
     # TODO: TableQueryResponse is not the correct type
     @classmethod
-    def _from_generated(cls, generated, **kwargs):
-        # type: (TableQueryResponse, Dict[str, Any]) -> TableItem
-        return cls(generated.table_name, **kwargs)  # type: ignore
+    def _from_generated(cls, generated, **kwargs):  # pylint: disable=unused-argument
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_models.py:479]: We probably can - however when I tried to do this, things broke. So I'd need to figure out why - and given that's a non-public implementation issue, I figure we can clean that up later.

File: sdk/tables/azure-data-tables/azure/data/tables/_table_service_client.py [hunk 2]
```diff
@@ -102,12 +102,12 @@ def from_connection_string(cls, conn_str, **kwargs):
 
     @distributed_trace
     def get_service_stats(self, **kwargs):
-        # type: (Any) -> Dict[str, Any]
+        # type: (Any) -> Dict[str, object]
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_table_service_client.py:105]: `Any` is a bit like `ignore`. Whereas `object` means there will be _something_.
It's a minor difference that probably wont do much :)

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 3]
```diff
@@ -104,37 +88,34 @@ class TableAnalyticsLogging(GeneratedLogging):
...
         self.version = kwargs.get("version", u"1.0")
         self.delete = kwargs.get("delete", False)
         self.read = kwargs.get("read", False)
         self.write = kwargs.get("write", False)
-        self.retention_policy = kwargs.get("retention_policy") or RetentionPolicy()
+        self.retention_policy = kwargs.get("retention_policy") or TableRetentionPolicy()
 
     @classmethod
-    def _from_generated(cls, generated):
+    def _from_generated(cls, generated, **kwargs):  # pylint: disable=unused-argument
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_models.py:104]: True - I added this back by accident when I broke one of the others. I will remove.

**Review by seankane-msft — APPROVED** [review]
:shipit:

**Pull request merged**